### PR TITLE
Implemented: toast for background process on pack action on in-progress page (#185)

### LINKED ISSUE #185: Display a toast for background process when performing pack action on in-progress page
## What is the motivation for adding/enhancing this feature?

Currently, when performing a pack action(single or in bulk) we are also generating some documents, but in the current scenario a loader is being displayed for the whole process.

## What are the acceptance criteria?

- [ ] A toast message should be displayed informing the user that some document generation is going on, and the loader should only be displayed until the api is pending.
- [ ] Toast should not be dismissed untill the documents are not generated, and the toast should have a dismiss button so user can dismiss the toast manually.

## Can you complete this feature request by yourself?

- [x] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by k2maan**
### Related Issues

Closes #185 

### Short Description and Why It's Useful

On performing the packing action on the in-progress page and selecting the option to generate shipping labels and/or packing slips, a toast will be displayed stating 'Order packed successfully. Document generation in process'.

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [ ] I read and followed [contribution rules](https://github.com/hotwax/ionic-bopis#contribution-guideline)

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -14,14 +14,28 @@ const hasError = (response: any) => {
...
+const showToast = async (message: string, canDismiss?: boolean, manualDismiss?: boolean, position?: string) => {
+  const config = {
+    message,
+    position: position ? position : 'bottom',
+  } as any
+
+  if (canDismiss) {
+    config.buttons = [
+      {
+        text: 'Dismiss',
```
Comment (ymaheshwari1) [inline src/utils/index.ts:26]: Add translation for static text

File: src/views/InProgress.vue [hunk 2]
```diff
@@ -395,19 +403,28 @@ export default defineComponent({
                 const resp = await OrderService.packOrders({
                   shipmentIds
                 });
-                if (resp.status === 200 && !hasError(resp)) {
-                  showToast(translate('Orders packed successfully'));
-                } else {
+                if (hasError(resp)) {
```
Comment (ymaheshwari1) [inline src/views/InProgress.vue:406]: When having error we direct the flow to catch block in that case the loader will not be dismissed.

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/utils/index.ts (hunk 1 above)
Comment (ymaheshwari1) [reply src/utils/index.ts:26]: Also, add translation in all the locale files in default en language.

File: src/views/InProgress.vue (hunk 2 above)
Comment (k2maan) [reply src/views/InProgress.vue:406]: Updated sir, handled in finally block

**Review by ymaheshwari1 — APPROVED** [review]

File: src/views/InProgress.vue [hunk 3]
```diff
@@ -318,30 +318,40 @@ export default defineComponent({
...
+
+                  if (data.includes('printPackingSlip') && data.includes('printShippingLabel')) {
+                    await OrderService.printShippingLabelAndPackingSlip(order.shipmentIds)
+                  } else if(data.includes('printPackingSlip')) {
+                    await OrderService.printPackingSlip(order.shipmentIds)
+                  } else if(data.includes('printShippingLabel')) {
+                    await OrderService.printShippingLabel(order.shipmentIds)
+                  }
+
+                  toast.dismiss()
```
Comment (adityasharma7) [inline src/views/InProgress.vue:342]: We should dismiss in catch block

File: src/views/InProgress.vue [hunk 4]
```diff
@@ -395,28 +405,38 @@ export default defineComponent({
...
-                if (data.includes('printPackingSlip') && data.includes('printShippingLabel')) {
-                  await OrderService.printShippingLabelAndPackingSlip(shipmentIds)
-                } else if(data.includes('printPackingSlip')) {
-                  await OrderService.printPackingSlip(shipmentIds)
-                } else if(data.includes('printShippingLabel')) {
-                  await OrderService.printShippingLabel(shipmentIds)
+                if (data.length) {
+                  // additional parameters for dismiss button and manual dismiss ability
+                  const toast: any = await showToast(translate('Order packed successfully. Document generation in process'), true, true)
+                  toast.present()
```
Comment (adityasharma7) [inline src/views/InProgress.vue:418]: We should dismiss in catch block

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/InProgress.vue (hunk 4 above)
Comment (k2maan) [reply src/views/InProgress.vue:421]: Updated sir

**Pull request merged**